### PR TITLE
Render image content type for Upstream #164

### PR DESCRIPTION
**Pull request by tiagralee**
- update restlib to support image content-type
- add related tests

### CODE REVIEW

File: restlib/restlib.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func unmarshal(resp *http.Response, body []byte, respStruct interface{}) (*HTTPR
 	}
 
 	contentType := resp.Header.Get("Content-Type")
-	if strings.Contains(contentType, "xml") {
+	if strings.Contains(contentType, "xml") || strings.Contains(contentType, "image") {
```
Comment (andrewemeryanz) [inline restlib/restlib.go]: @tiagralee Can you check with @suipinglu that this solves his issue.

His issue tags a different line than the one touched in this pull request:
https://github.com/anz-bank/sysl-go/blob/master/restlib/restlib.go#L82

**Review by andrewemeryanz — CHANGES_REQUESTED** [review]
Check with owner of issue and add `Closes #164` to the commit message.

File: restlib/restlib_test.go [hunk 2]
```diff
@@ -65,6 +66,19 @@ func TestUnmarshalEmptyBodyOK(t *testing.T) {
 	require.Nil(t, result.Response)
 }
 
+func TestUnmarshalImageContentType(t *testing.T) {
+	header := map[string][]string{
+		"Content-Type": {"image/png"},
+	}
+	result, err := unmarshal(&http.Response{Header: header}, []byte(okJSON), &OkType{})
```
Comment (andrewemeryanz) [inline restlib/restlib_test.go]: The logic is fine but can you change the example from: 
```
[]byte(okJSON)
``` 

into something like 
```
var image = [...]byte { 0, 1 }
```

I was a bit confused when the test is for an image response and the response contained a json string.

**Review by andrewemeryanz — CHANGES_REQUESTED** [review]

File: restlib/restlib.go [hunk 3]
```diff
@@ -57,9 +58,12 @@ func unmarshal(resp *http.Response, body []byte, respStruct interface{}) (*HTTPR
 	}
 
 	contentType := resp.Header.Get("Content-Type")
-	if strings.Contains(contentType, "xml") {
+	switch {
+	case strings.Contains(contentType, "xml"):
 		respStruct = string(body)
-	} else {
+	case strings.Contains(contentType, "image"):
+		respStruct = base64.StdEncoding.EncodeToString(body)
```
Comment (andrewemeryanz) [inline restlib/restlib.go]: We shouldn't be Base64 encoding this. A byte array should suffice.

**Review by andrewemeryanz — CHANGES_REQUESTED** [review]

File: restlib/restlib.go [hunk 4]
```diff
@@ -49,14 +49,16 @@ func unmarshal(resp *http.Response, body []byte, respStruct interface{}) (*HTTPR
 		return makeHTTPResult(resp, body, nil), nil
 	}
 
+	contentType := resp.Header.Get("Content-Type")
+
 	e := reflect.ValueOf(respStruct).Elem()
-	if e.Kind() == reflect.String {
+	kind := e.Kind()
+	if kind == reflect.String || (kind == reflect.Slice && strings.Contains(contentType, "image")) {
```
Comment (andrewemeryanz) [inline restlib/restlib.go]: Is it possible to generalise this check to byte arrays? 
```
Current:  Is this a slice and an image.
Proposed: Is this a slice of bytes.
```

Comment (tiagralee) [reply restlib/restlib.go]: Good catch, the "text/plain" for ApiDoc is slice as well, I'll have a look if it can further check the type in the slice.

**Review by andrewemeryanz — APPROVED** [review]
Great.

**Pull request merged**